workspaces/local-cli/src/shared/local-cli-task-runner.ts

```diff
@@ -241,6 +234,41 @@ export class LocalCliTaskRunner implements IOpticTaskRunner {
     developerDebugLogger(`api base url: ${apiBaseUrl}`);
     const cliClient = new Client(apiBaseUrl);
 
+    ////////////////////////////////////////////////////////////////////////////////
+    const spectacle = new LocalCliSpectacle(apiBaseUrl, opticEngine);
+    const idQuery = await spectacle.query<any>({
+      query: `{
+        metadata {
+          id
+        }
+      }`,
+      variables: {},
+    });
+    const specId = idQuery?.data?.metadata?.id;
+
+    const batchCommitQuery = await spectacle.query<any>({
+      query: `{
+        batchCommits {
+          createdAt
+        }
+      }`,
+      variables: {},
+    });
+
+    const createdAt = batchCommitQuery?.data?.batchCommits
+      ?.map((commit: any) => commit?.createdAt)
+      ?.sort()?.[0]?.createdAt;
```

Comment:
I think batchCommits comes back as a numeric string? Sort will do an alphanumeric sort on strings, which i don't believe is what we want

Reply:
asdasdasd I meant to map to `new Date()`

Reply:
You'll need to do something like https://stackoverflow.com/questions/10123953/how-to-sort-an-object-array-by-date-property:

```
array.sort(function(a,b){
  // Turn your strings into dates, and then subtract them
  // to get a value that is either negative, positive, or zero.
  return new Date(b.date) - new Date(a.date);
});
```

and since you are already mapping this, you don't need `?.createdAt` at the end

Reply:
Yup good point. I changed it to 

```
const createdAt = batchCommitQuery?.data?.batchCommits
      ?.map((commit: any) => new Date(commit?.createdAt))
      ?.sort()?.[0];
```

Dates appear to be comparable like regular numbers, so regular old `.sort()` should do the trick

Reply:
We want an ascending sort, not a descending sort - sort by default looks like it gives a descending sort?